docs/rfq-and-price-oracle.md

```diff
@@ -0,0 +1,233 @@
+
+**Example 2: Paying an invoice with MPP**:
+
+Multi-Part Payments (MPP) allow a single payment to be split up into multiple
+parts/paths, resulting in multiple HTLCs per payment.
+
+If a user wants to pay an invoice over `1,000,000` satoshi, that would be
+equivalent to `52,631.5789` USD cents.
+If the user were to pay this amount in a single HTLC, they would send `52,632`
+USD cents (need to round up to satisfy integer asset amount and invoice minimum
+amount requirements), overpaying by `0.4211` cents.
+
+If the user's wallet decided to split up the payment into 16 parts for example,
+then each part would correspond to `3289.4737` cents. To satisfy the integer
+asset amount and invoice minimum amount requirement, each of the 16 HTLCs would
+send out `3290` cents. That's a full `8.4211` cents of overpayment.
+
+## What precision should I choose when minting an asset for the Lightning Network?
+
+To address the issue of rounding up when splitting payments or representing
+small satoshi amounts as asset units, an issuer of assets should use a high
+enough value for `decimal_display` when minting.
+
+**But what is a good value for `decimal_display`?**
+
+We recommend to use a `decimal_display` value of `6` for currencies which
+use a smaller subunit with two decimal places (such as cents for USD or EUR,
+penny for GBP and so on).
+
+For currencies without smaller units (for example JPY or VND), a
+`decimal_display` value of `4` is recommended.
+
+# RFQ
+
+The RFQ system is responsible for acquiring real-time price quotes for
+converting between asset units and satoshi (both directions) or between
+different asset types.
+
+There are two main user stories (as seen from the point of view of the wallet
+end user):
+
+1. Sending out assets: The user wants to pay a Lightning Network invoice that
+   is denominated in satoshi. The user only has assets in their wallet, they
+   (or their wallet software) want to find out how many asset units they need to
+   send in order to satisfy the invoice amount in satoshi.
+2. Receiving assets: The user wants to get paid in a specific asset. The user
+   only knows about the asset, so they (or their wallet software) want to find
+   out what the asset amount corresponds to in satoshi.
+
+## Sell order
+
+The sell order covers the first user story: The user wants to pay a
+satoshi-denominated invoice with assets. The end result is that the user sells
+some of their assets in their channel to the RFQ peer (edge node), requesting
+them to forward satoshi to the network.
+
+Formal definition (TODO: refine):
+- Use case: sending assets as a payment, selling `buxx` for `msat`
+- User query: `Q = how many out_asset units for in_asset amount?` (how many
+  `buxx` do I need to seel to pay this payment denominated in `msat`?)
+- `out_asset`: `buxx` (user sells `buxx` asset to RFQ peer)
+- `in_asset`: `msat` (user "receives" `msat` from RFQ peer, which are then
```

Comment:
Does the in/out asset delineation match the current RFQ spec/code? Thinking about it anew, perhaps we should frame it as relative to the edge node: the edge node? 

In that case, it would be the reverse here:
  * The edge node receives `buxx` (`in_asset`)
  * The edge node sends out `msat` (`the out_asset)

A similar reframing might be useful in the section below re receiving.

Reply:
Both the current RFQ and RPC code in `tapd` as well as the overhauled bLIP code in https://github.com/Roasbeef/blips/pull/4 always define things from the point of view of the **wallet end user** and not the edge node.
I'd like to avoid flipping that around as that would require a lot of changes (and potentially lead to more confusion). So I made the direction very clear in the `# RFQ` section.